CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Contributing
+
+Thank you for contributing to KGraphQL!
+All contributions are welcome, feel free to open issues and PRs.
+
+## Getting Started
+
+To perform a local build you only need to have JDK 11 installed and invoke
+
+```bash
+./gradlew build
+```
+
+You can verify your changes using
+
+```bash
+./gradlew check
+```
+
+## Creating a Pull Request
+
+Pull requests are the best way of getting your changes into the code base.
+Before opening a pull request for non-trivial changes, it might be good to open an issue for discussion first to
+avoid unnecessary work, although this is not required.
+
+To support reviewing your changes please
+
+- Target the `main` branch
+- Submit **one** pull request per bug or feature to keep the changes small and concise
+- Follow existing code style and conventions
+- Add automated tests when possible
+- Structure your changes in as little _meaningful_ commits as possible. Don't be shy to force-push if needed.
+
+## Writing Commit Messages
+
+Commit messages matter. Please have a look at [How to Write a Git Commit Message](https://cbea.ms/git-commit/) for a
```

Comment:
To be discussed: https://www.conventionalcommits.org/en/v1.0.0/

Reply:
We as maintainers could just do that internally by squashing PRs into one commit and name that one commit accordingly.

This _may_ be easier for possible contributors.

Reply:
I like the idea, contributions should definitely be as easy as possible.

I'm thinking wether we should simply require a PR title+description _that make sense_ and do the (conventional) commit stuff during merge. Only the original author can tell what their intention and thought process was, so this IMHO needs to be a part of the PR itself, but which prefixes to use is maybe more of an implementation detail.

Reply:
I like the idea to motivate contributors to name the title according to conventional commits.

IMO it would be _optimal_ if the PR title would reflect the commit messages first line and the PR description the commit message body.

From a format and wording point of view.

Reply:
Updated the description with less focus on "this is what you _have to_ do" and more focus on "_ideally_ you would do this".